mcs/class/System/System.Net/HttpWebRequest.cs

```diff
@@ -828,7 +828,7 @@ void AddRange (string rangeSpecifier, long from, long to)
 		WebOperation SendRequest (bool redirecting, BufferOffsetSize writeBuffer, CancellationToken cancellationToken)
 		{
 			lock (locker) {
-				WebConnection.Debug ($"HWR SEND REQUEST: Req={ID} requestSent={requestSent} redirecting={redirecting}");
+				WebConnection.Debug ($"HWR SEND REQUEST: Req={ID} requestSent={requestSent} actualUri={actualUri} redirecting={redirecting}");
```

Comment:
Minor issue but won't $"..." generate garbage and burn cycles string formatting even if debug messages are suppressed? Wouldn't it be better to do "... {0} ...", x style instead

Reply:
In general yes but `WebConnection.Debug()` is marked as `[Conditional ("MONO_WEB_DEBUG")]` so the compiler will not even emit it into the output assembly.